### PR TITLE
remove using object.keys in iterations + cache previous key index + remove new instances

### PR DESCRIPTION
**Pull request by thomasbiang**
Fixes #

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/playcanvas-anim.js [hunk 1]
```diff
@@ -1,3 +1,6 @@
+// 1. remove object.keys
+// 2. remove new
+// 3.
```
Comment (willeastcott) [inline src/playcanvas-anim.js]: Maybe remove these comments. 😄

**Comment by thomasbiang** [comment]
remove comments
cache previous key index in eval functions

**Comment by willeastcott** [comment]
Would you like me to review and merge this PR yet? 😄

**Comment by thomasbiang** [comment]
Hi Will,

Sure.  Commission till de9e966 is for removing Object.keys in iterations. Commission till 32debfc is for caching previous key index in curve evaluation. The rest is removing new instances in evaluation. 
They are tested with several gltf/glb files with animations (both LINEAR and CUBIC).

**Comment by willeastcott** [comment]
OK, I've also tested it and it seems faster with fewer garbage collection events. Impressive. I'll continue profiling to see if there are other opportunities for optimization. Thanks!! 😄

**Comment by willeastcott** [comment]
Actually, I did just find a problem. Check out this line:

https://github.com/playcanvas/playcanvas-gltf/blob/master/src/playcanvas-anim.js#L1856

This calls this function with one parameter:

https://github.com/playcanvas/playcanvas-gltf/blob/master/src/playcanvas-anim.js#L1334

Since the ```cacheValue``` argument is not supplied, this line throws an exception because ```snapshot``` is undefined:

https://github.com/playcanvas/playcanvas-gltf/blob/master/src/playcanvas-anim.js#L1336

**Pull request merged**